Dockerfile

```diff
@@ -0,0 +1,30 @@
+#
+# Node JS Alpine image
+#
+
+FROM gliderlabs/alpine:3.4
+
+# Install root filesystem
+ADD ./rootfs /
+RUN echo -ne "Alpine Linux 3.4 image. (`uname -rsv`)\n" >> /root/.built
+
+# Node and NPM version
+ENV NODE_VERSION=6.3.1 NPM_VERSION=3.10.5
+
+# Install Node and NPM
+RUN apk update && apk-install curl bash tree make gcc g++ python linux-headers paxctl libgcc libstdc++ && \
+  curl -sSL https://nodejs.org/dist/v${NODE_VERSION}/node-v${NODE_VERSION}.tar.gz | tar -xz && \
```

Comment:
I wanted to get the smallest image possible, hence the less extra things. That's why I download the source and build tools and build it, then remove the build dependencies. 
Rather than using nvm, I'd change the `NODE_VERSION` and build another image for this version. 
The output are tagged images, like `smooch/node:6` (equal to `smooch/node:latest`), and I haven't yet built `smooch/node:5` but you get the idea: only build the smallest file, and tag it with the version used.